// Copyright 2015 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Protocol buffer definition for a delta file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package history_report;

message DeltaFileEntry {
  // Next ID to use: 4

  // |seq_no| is an auto increment sequence number, used as the key.
  optional int64 seq_no = 1;

  // |type| can only be "add" or "del".
  optional string type = 2;

  optional string url = 3;
}

message UsageReport {
  // Next ID to use: 4

  // |id| is derived from the URL, and the length doesn't exceed 256 chars.
  // DeltaFileEntry and UsageReport can be joined by |id| this way.
  optional string id = 1;

  optional int64 timestamp_ms = 2;

  // Whether the navigation is PAGE_TRANSITION_TYPED and not redirected.
  optional bool typed_visit = 3;
}
